src/reynir_correct/pattern.py

```diff
@@ -178,15 +178,37 @@ def __init__(self, ann: List[Annotation], sent: Sentence) -> None:
     def get_wordform(self, lemma: str, cat: str, variants: Iterable[str]):
         """ Get correct wordform from BinPackage, 
         given a set of variants """
-
-        # Get rid of argument variants in verbs:
-        realvars = [v for v in variants if not v.isdigit() and v not in SKIPVARS]
+        cases = ["nf", "þf", "þgf", "ef"]
+        if cat == "so":
+            # Get rid of argument variants in verbs:
+            realvars: Iterable[str]  = [v for v in variants if not v.isdigit() and v not in SKIPVARS]
+            if not "lh" in realvars:
+                realvars = [v for v in realvars if v not in cases]
+        else:
+            realvars= variants
         wordforms = BIN.lookup_variants(lemma, cat, tuple(realvars))
         if not wordforms:
             return ""
         # Can be many possible word forms, want the first one in most cases
         return wordforms[0].bmynd
 
+    def wrong_subtree(self, match: SimpleTree, so: SimpleTree) -> bool:
+        iptree = match.first_match("IP")
+        if not iptree:
+            return False
+    
+        for x in iptree.descendants:
+            if not x.is_terminal and x.tag == "IP":
+                if self.is_subtree(so, x):
+                    return True
+        return False
+
+    def is_subtree(self, first: SimpleTree, other: SimpleTree) -> bool:
+        """ Returns True if first tree is a subtree of the second one """
+        a, b = first.span
+        c, d = other.span
+        return set(range(a, b+1)).issubset(set(range(c, d+1)))
```

Comment:
Simpler: return (c >= a) and (d <= b)  (this is true if c,d is contained within a,b)

Reply:
Done